Implement bitsize pragma for bitfields

tmm1 opened the pull request:
cc https://github.com/nim-lang/c2nim/pull/53

``` nim
type
  bits* = object
    flag* {.bitsize: 1.}: cint
    opts* {.bitsize: 4.}: cint
```

``` c
struct  bits95004  {
int flag:1;
int opts:4;
};
```

Varriount commented:
What does sizeof() do when faced with a type containing bitfields, or a bitfield member?

Araq commented:
@Varriount The compiler handles that appropriately already since sizeof is not support at compile-time for objects. Instead the sizeof is delegated to the C code.

Araq commented:
the patch is really good but using BiggestInt for this seems a bit wasteful. Is  sizeof(TSym) still the same after this patch?

Araq commented:
(sorry, pressed the wrong button...)

tmm1 commented:
> using BiggestInt for this seems a bit wasteful

Agreed, I wonder what the C spec says about a max bitsize. Also wonder if we should disallow the bitsize pragma unless used with `cint` or `cuint`

tmm1 commented on tests/pragmas/tbitsize.nim:
```diff
@@ -0,0 +1,7 @@
+    opts* {.bitsize: 4.}: cint
+
+var b: bits
+echo b.flag
```
Is there an easy way to make assertions against the generated C code here, so I can verify the bitfield is being generated as expected?

tmm1 commented:
I've switched to a simple `int`. All the other case branches of TSymKind had two fields already, so I believe the size of the structure should not have changed.

I added some better tests, and finished up the companion c2nim PR https://github.com/nim-lang/c2nim/pull/53 to generate compatible files. Both PRs are ready for review/merge.

Araq commented on tests/pragmas/tbitsize.nim:
```diff
@@ -0,0 +1,18 @@
+
+var
+  b: bits
+
```
You should assert the sizeof() here to see it ended up in the generated C code. There is also a ccodeGenCheck but I forgot its syntax.

tmm1 replied:
On my machine `sizeof(bits) == 4`, is it safe to assume this will be consistent across platforms?

The `assert (b.flag = 2) == 0` below should be a pretty reliable test since it ensures the field is only one bit long.

tmm1 replied:
I've added a `ccodeCheck` assertion.

Araq commented:
Ok, that's fine. Now document it please. Add it to news.txt and to docs/pragmas.txt or somewhere it belongs.

tmm1 commented:
> Ok, that's fine. Now document it please. Add it to news.txt and to docs/pragmas.txt or somewhere it belongs.

Done.

tmm1 commented:
Thanks! Can you merge https://github.com/nim-lang/c2nim/pull/53 as well?
